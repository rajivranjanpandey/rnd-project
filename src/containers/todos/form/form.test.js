import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoForm from './index';
const props = {
    addTodo: jest.fn()
}
describe('TodoForm', () => {
    beforeEach(() => {
        jest.clearAllMocks()
    })
    it('should take a snapshot', () => {
        const { asFragment } = render(<TodoForm />);
        expect(asFragment(<TodoForm />)).toMatchSnapshot();
    });
    it('input value updates on user input', async () => {
        const inputValue = 'write test cases';
        const { queryByTestId } = render(<TodoForm {...props} />);
        const input = queryByTestId('todo-input');
        expect(input).toBeInTheDocument();

        fireEvent.change(input, { target: { value: inputValue } });
        expect(input).toHaveValue(inputValue);
    })
    it('todo gets saved', async () => {
        const inputValue = 'write test cases';
        const { queryByTestId, queryByText } = render(<TodoForm {...props} />);
        const input = queryByTestId('todo-input');
        expect(input).toBeInTheDocument();

        fireEvent.change(input, { target: { value: inputValue } });
        const saveButton = queryByText('Save');
        expect(saveButton).toBeInTheDocument();
        fireEvent.click(saveButton);
        expect(props.addTodo).toBeCalled();
    })
});

