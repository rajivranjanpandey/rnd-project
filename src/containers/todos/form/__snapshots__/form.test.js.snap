// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root todo_input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="outlined-controlled"
      id="outlined-controlled-label"
    >
      Your inputs ... 
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="todo-input"
        id="outlined-controlled"
        type="text"
        value=""
      />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Save
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            Your inputs ... 
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
